### PR TITLE
Ensure correct shutdown order when session is closed

### PR DESCRIPTION
**Pull request by thallgren**
The DNS service needs the TUN-device to be alive during shutdown, and
closing the TUN-device is what unblocks the routerWorker's read on that
device. This commit ensures that the DNS-worker dies first, followed by
a close down of the TUN-device (which implicitly ends the routerWorker).

### CODE REVIEW

File: pkg/client/daemon/session.go [hunk 1]
```diff
@@ -426,42 +426,47 @@ func (s *session) run(c context.Context) error {
...
-	}
-	if atomic.CompareAndSwapInt32(&s.closing, 1, 2) {
-		cc := dcontext.WithoutCancel(c)
-		for _, np := range s.curStaticRoutes {
-			err := s.dev.RemoveStaticRoute(cc, np)
-			if err != nil {
-				dlog.Warnf(c, "error removing route %s: %v", np, err)
-			}
+func (s *session) stop(c context.Context) {
+	atomic.StoreInt32(&s.closing, 1)
```
Comment (maxlegault) [inline pkg/client/daemon/session.go:445]: Is it still valuable to keep the closing state if it's not compared anymore?

**Review by donnyyung — APPROVED** [review]
LGTM!

File: pkg/client/daemon/session.go (hunk 1 above)
Comment (thallgren) [reply pkg/client/daemon/session.go:445]: There's no point in comparing now. The `stop` function is the only one storing the value and it's guaranteed to only be called once when the session ends.

Comment (thallgren) [reply pkg/client/daemon/session.go:445]: And the value is used by the router.

Comment (maxlegault) [reply pkg/client/daemon/session.go:445]: Ah got it, I hadn't looked at the router.go file. Thanks

**Pull request merged**